src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java

```diff
@@ -316,11 +317,12 @@ public void exit(AbruptExitException exception) {
             "CommandEnvironment needs its options provider to have ClientOptions loaded.");
 
     this.clientEnv = makeMapFromMapEntries(clientOptions.clientEnv);
-    this.commandId = computeCommandId(commandOptions.invocationId, warnings);
+    this.commandId = computeCommandId(commandOptions.invocationId, warnings, attemptNumber);
+    // Use a fixed ID across cache eviction retries.
```

Comment:
nit: move this comment to the call site where we set `buildIdOverride` to a non-null value.